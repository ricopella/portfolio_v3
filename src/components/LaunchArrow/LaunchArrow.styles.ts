import styled from '@emotion/styled'
import { COLORS } from '../../styles/variables'
import { OutboundLink } from 'gatsby-plugin-google-analytics'

const LaunchArrowWrapper = styled(OutboundLink)`
  align-items: center;
  background-color: ${COLORS.terminalBackground};
  border-radius: 0.3125rem;
  display: grid;
  grid-template-columns: 1fr;
  height: 1.5rem;
  justify-content: center;
  position: relative;
  width: 1.5rem;
  border: none;

  &:hover {
    cursor: pointer;
    transform: scale(1.1);
  }
`

const LaunchArrowInner = styled.div`
  border-bottom-color: transparent;
  border-left-color: transparent;
  border-style: solid;
  border-width: 0.375rem;
  box-sizing: border-box;
  color: ${COLORS.white};
  display: inline-block;
  height: 0;
  position: absolute;
  right: 0.3125rem;
  top: 0.3125rem;
  vertical-align: middle;
  width: 0;
  &:after,
  &:before {
    content: '';
    box-sizing: border-box;
  }
  &:before {
    right: 0;
    top: -0.1875rem;
    position: absolute;
    height: 0.25rem;
    box-shadow: inset 0 0 0 2rem;
    transform: rotate(-45deg);
    width: 0.75rem;
    transform-origin: right top;
  }
`

export default {
  LaunchArrowInner,
  LaunchArrowWrapper,
}
