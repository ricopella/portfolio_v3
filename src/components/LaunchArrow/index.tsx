import React, { FC } from 'react'
import Styled from './LaunchArrow.styles'
import { LaunchArrowProps } from '../../types'

const LaunchArrow: FC<LaunchArrowProps> = ({ href }) => (
  <Styled.LaunchArrowWrapper
    href={href}
    rel="noopener noreferrer"
    target="_blank"
    title={href}
  >
    <Styled.LaunchArrowInner />
  </Styled.LaunchArrowWrapper>
)

export default LaunchArrow
