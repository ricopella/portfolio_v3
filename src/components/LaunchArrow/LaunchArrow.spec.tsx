import LaunchArrow from './index'
import React from 'react'
import renderer from 'react-test-renderer'

describe('<LaunchArrow />', () => {
  it('should match LaunchArrow snapshot', () => {
    const internalTree = renderer
      .create(<LaunchArrow href="foo.com" />)
      .toJSON()

    expect(internalTree).toMatchSnapshot()
  })
})
