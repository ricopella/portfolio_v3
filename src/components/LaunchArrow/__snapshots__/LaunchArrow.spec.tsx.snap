// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LaunchArrow /> should match LaunchArrow snapshot 1`] = `
<a
  className="css-c1h6nt"
  href="foo.com"
  onClick={[Function]}
  rel="noopener noreferrer"
  target="_blank"
  title="foo.com"
>
  <div
    className="css-3nfvs0"
  />
</a>
`;
